parser.js

```diff
@@ -24,77 +23,64 @@ function buildRuleset(name, rules) {
-
-const imageRules = buildRuleset('image', [
-  ['meta[property="og:image:secure_url"]', node => node.element.content],
-  ['meta[property="og:image:url"]', node => node.element.content],
-  ['meta[property="og:image"]', node => node.element.content],
-  ['meta[property="twitter:image"]', node => node.element.content],
-  ['meta[name="thumbnail"]', node => node.element.content],
-]);
-
-const descriptionRules = buildRuleset('description', [
-  ['meta[property="og:description"]', node => node.element.content],
-  ['meta[name="description"]', node => node.element.content],
-]);
-
-const typeRules = buildRuleset('type', [
-  ['meta[property="og:type"]', node => node.element.content],
-]);
-
-
 const metadataRules = {
-  description: descriptionRules,
-  icon_url: iconRules,
-  image_url: imageRules,
-  keywords: keywordsRules,
-  title: titleRules,
-  type: typeRules,
-  url: canonicalUrlRules
+  description: [
+    ['meta[property="og:description"]', node => node.element.content],
+    ['meta[name="description"]', node => node.element.content],
+  ],
+  icon_url: [
+    ['link[rel="apple-touch-icon"]', node => node.element.href],
+    ['link[rel="apple-touch-icon-precomposed"]', node => node.element.href],
+    ['link[rel="icon"]', node => node.element.href],
+    ['link[rel="fluid-icon"]', node => node.element.href],
+    ['link[rel="shortcut icon"]', node => node.element.href],
+    ['link[rel="Shortcut Icon"]', node => node.element.href],
+    ['link[rel="mask-icon"]', node => node.element.href],
+  ],
+  image_url: [
+    ['meta[property="og:image:secure_url"]', node => node.element.content],
+    ['meta[property="og:image:url"]', node => node.element.content],
+    ['meta[property="og:image"]', node => node.element.content],
+    ['meta[property="twitter:image"]', node => node.element.content],
+    ['meta[name="thumbnail"]', node => node.element.content],
+  ],
+  keywords: [
+    ['meta[name="keywords"]', node => node.element.content],
+  ],
+  title: [
+    ['meta[property="og:title"]', node => node.element.content],
+    ['meta[property="twitter:title"]', node => node.element.content],
+    ['meta[name="hdl"]', node => node.element.content],
+    ['title', node => node.element.text],
+  ],
+  type: [
+    ['meta[property="og:type"]', node => node.element.content],
+  ],
+  url: [
+    ['meta[property="og:url"]', node => node.element.content],
+    ['link[rel="canonical"]', node => node.element.href],
+  ],
 };
 
-
 function getMetadata(doc, rules) {
   const metadata = {};
   const ruleSet = rules || metadataRules;
 
   Object.keys(ruleSet).map(metadataKey => {
     const metadataRule = ruleSet[metadataKey];
-    metadata[metadataKey] = typeof metadataRule === 'function' ?
-      metadataRule(doc) :
-      getMetadata(doc, metadataRule);
+
+    if(Array.isArray(metadataRule)) {
+      metadata[metadataKey] = buildRuleset(metadataKey, metadataRule)(doc);
+    } else {
+      metadata[metadataKey] = getMetadata(doc, metadataRule);
+    }
```

Comment:
It's a weird difference.  Where previously we were looking for typeof function because we had already applied buildRuleset and so each 'rule' was a callable.  But now we've pushed that off, and so we're looking for arrays which are lists of rules, and then we wrap them in buildRuleset and apply them, or recurse deeper.